src/Text/Pandoc/Writers/LaTeX.hs

```diff
@@ -1423,24 +1423,32 @@ citeCommand c p s k = do
   args <- citeArguments p s k
   return $ literal ("\\" <> c) <> args
 
+citeArgumentsList :: PandocMonad m
+              => [Inline] -> [Inline] -> [Text] -> LW m (Doc Text)
+citeArgumentsList _ _ [] = return empty
+citeArgumentsList pfxs sfxs (cid:ids) = do  
+      pdoc <- inlineListToLaTeX pfxs
+      sdoc <- inlineListToLaTeX sfxs' 
+      if null ids then  
+            return $ (optargs pdoc sdoc) <> braces (literal cid)
+      else return $ (optargs pdoc sdoc) <> 
+              (braces (literal (T.intercalate "," (cid:ids))))
+      where sfxs' = stripLocatorBraces $ case sfxs of
```

Comment:
I'm not sure this conditional is needed, since when ids is null, `T.intercalate "," (cid:ids) == cid`, right?
So the else clause by itself should be all you need.